packages/rs-sdk/src/sdk.rs

```diff
@@ -539,11 +542,14 @@ impl Sdk {
         }
     }
 
+    // TODO: Move to settings
     /// Indicate if the sdk should request and verify proofs.
     pub fn prove(&self) -> bool {
         self.proofs
     }
 
+    // TODO: If we remove this setter we don't need to use ArcSwap.
+    //   It's good enough to set Context once when you initialize the SDK.
```

Comment:
@shumkov this is not so simple - to retrieve data contracts, ContextProvider requires Sdk. It creates circular dependency, where ContextProvider needs Sdk to initialize, and Sdk needs ContextProvider. 

See https://github.com/dashpay/platform/blob/e7213a7affd5669d351a549591a01da821fd28e8/packages/rs-sdk/examples/read_contract.rs#L95-L96 how it's done. Not sure if this is optimal solution, but it works.